src/metatensor/models/cli/train_model.py

```diff
@@ -153,6 +156,13 @@ def _train_model_hydra(options: DictConfig) -> None:
     generator = torch.Generator()
```

Comment:
Because we need a generator for the split of the dataset.

https://pytorch.org/docs/stable/data.html#torch.utils.data.random_split

If you have a better option I am happy to discuss it.

Reply:
Ideally we should not use the global number generator, and always pass a `generator` instance around (that would also help a lot with reproducibility). In practise this might be a bit too much for now =/

Reply:
Ok!

Reply:
> In practise this might be a bit too much for now =/

Unfortunately, I agree.